### PR TITLE
Prepare for prow adoption

### PR DESCRIPTION
**Pull request by jlpadilla**
Issue: https://github.com/open-cluster-management/backlog/issues/9995

### Changes
- Add `Dockerfile.prow`
- Add `Makefile.prow`
- Update Dockerfile to be consistent with Dockerfile.prow
- Add DCO

### CODE REVIEW

File: Dockerfile.prow [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
+
+# Copy the go source
+COPY main.go main.go
+COPY api/ api/
+COPY controllers/ controllers/
+
+# Build
+RUN CGO_ENABLED=0 go build -a -o manager main.go
```
Comment (mqhackett) [inline Dockerfile.prow:20]: Should we add `GOGC=25`? We don't usually have memory problems on this repo -- just for consistency

File: .gitignore [hunk 2]
```diff
@@ -82,4 +82,5 @@ build-harness/
 !vbh/.build-harness-vendorized
 !vbh/build-harness
 !vbh/build-harness-extensions
+testbin/
```
Comment (mqhackett) [inline .gitignore:85]: What does `testbin/` come from?

File: Dockerfile.prow (hunk 1 above)
Comment (jlpadilla) [reply Dockerfile.prow:20]: Let's keep it simple. We only see problems in the aggregator project and it seems to be better with newer versions of go and libraries, so maybe we can eventually remove from all repos.

File: .gitignore (hunk 2 above)
Comment (jlpadilla) [reply .gitignore:85]: When I run `make unit-test` locally it generates this directory with some files.

**Review by mqhackett — APPROVED** [review]

**Pull request merged**